src/Workspaces/MSBuild/Test/RpcTests.cs

```diff
@@ -203,6 +215,18 @@ public async Task CancelledTaskDoesNotLeakRequest()
             Assert.Equal(0, rpcPair.Client.GetTestAccessor().GetOutstandingRequestCount());
         }
 
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/77040")]
+        public async Task RequestThatClosesServerDoesNotThrow()
+        {
+            await using var rpcPair = new RpcPair();
+            rpcPair.Server.AddTarget(new ObjectWithMethodThatShutsDownServer(rpcPair.Server));
+
+            // Invoke the method, and ensure we don't get exceptions back due to the shutdown if the pipe closed too early.
+            // This is not guaranteed to catch any bug, since any bug is fundamentally a race, but it at least ensures we aren't always broken.
```

Comment:
I wasn't able to get it to fail locally, but the test should ensure we don't break this too badly.